spec/board_spec.rb

```diff
@@ -18,4 +20,46 @@
             expect(@board.cells).to be_a(Hash)
         end
     end
+
+    describe '#validating coordinates' do
+        it 'can make sure coordinates are the correct length for the ship' do
+
+            expect(@board.valid_placement?(@cruiser, ["A1", "A2"])).to eq(false)
+            expect(@board.valid_placement?(@submarine, ["A2", "A3", "A4"])).to eq(false)
+        end
+   
+        it 'can tell if coordinates are consecutive' do
+
+            expect(@board.valid_placement?(@cruiser, ["A1", "A2", "A4"])).to eq(false)
+            expect(@board.valid_placement?(@submarine, ["A1", "C1"])).to eq(false)
+            expect(@board.valid_placement?(@cruiser, ["A3", "A2", "A1"])).to eq(false)
+            expect(@board.valid_placement?(@submarine, ["C1", "B1"])).to eq(false)
+        end
+
+        it 'can tell if coordinates are diaganol' do
+
+            expect(@board.valid_placement?(@cruiser, ["A1", "B2", "C3"])).to eq(false)
+            expect(@board.valid_placement?(@submarine, ["C2", "D3"])).to eq(false)
+        end
+
+        it 'can place a ship if all parameters are correctly met' do
+
+            expect(@board.valid_placement?(@submarine, ["A1", "A2"])).to eq(true)
+            expect(@board.valid_placement?(@cruiser, ["B1", "C1", "D1"])).to eq(true)
+        end
+    end
```

Comment:
I think we could take these two tests and add them to one of the above placement tests. I think the next @board.place test verifies the ability to place a ship if the parameters are met.